### PR TITLE
Integration tests

### LINKED ISSUE #9: Fix lock (key - "kmm-asset-lock") not obtained
After failing to create a control plane, the kmm terminates cleanly (with error) but lock is NOT cleaned up OR can't be re-obtained:
```
Apr 19 16:03:19 ip-172-31-4-32.eu-west-2.compute.internal docker[24422]: time="2017-04-19T16:03:19Z" level=info msg="Saving:"/etc/kubernetes/scheduler.conf""
Apr 19 16:03:19 ip-172-31-4-32.eu-west-2.compute.internal docker[24422]: [apiclient] Created API client, waiting for the control plane to become ready
Job for kmm.service failed because the control process exited with error code.
See "systemctl status kmm.service" and "journalctl -xe" for details.
Apr 19 16:04:37 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: kmm.service: Start-pre operation timed out. Terminating.
Apr 19 16:04:37 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: kmm.service: Control process exited, code=exited status=2
Apr 19 16:04:37 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: Failed to start Kubernetes Multi-master.
Apr 19 16:04:37 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: kmm.service: Unit entered failed state.
Apr 19 16:04:37 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: kmm.service: Failed with result 'exit-code'.
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: kmm.service: Service hold-off time over, scheduling restart.
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: Stopped Kubernetes Multi-master.
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal systemd[1]: Starting Kubernetes Multi-master...
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:04:47Z" level=info msg="Cloud Provider Initialized ["aws"]"
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:04:47Z" level=info msg="Getting "kmm-asset-key" key value..."
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:04:47Z" level=info msg="Assets not present in etcd...
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: "
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:04:47Z" level=info msg="Transaction didn't succeed - we didn't create lock!"
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:04:47Z" level=info msg="Lock allready created..."
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:04:47Z" level=info msg="Getting "kmm-asset-lock" key value..."
Apr 19 16:04:47 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:04:47Z" level=info msg="Lock (key - "kmm-asset-lock") not obtained, TTL exists:"2017-04-19T16:05:13Z""
Apr 19 16:05:07 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:05:07Z" level=info msg="Getting "kmm-asset-key" key value..."
Apr 19 16:05:07 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:05:07Z" level=info msg="Assets not present in etcd...
Apr 19 16:05:08 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: "
Apr 19 16:05:08 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:05:08Z" level=info msg="Transaction didn't succeed - we didn't create lock!"
Apr 19 16:05:08 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:05:08Z" level=info msg="Lock allready created..."
Apr 19 16:05:08 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:05:08Z" level=info msg="Getting "kmm-asset-lock" key value..."
Apr 19 16:05:08 ip-172-31-4-32.eu-west-2.compute.internal docker[25216]: time="2017-04-19T16:05:08Z" level=info msg="Lock (key - "kmm-asset-lock") not obtained, TTL exists:"2017-04-19T16:05:13Z""
```

### LINKED ISSUE #40: Add crucial integration tests
Add integration tests to ensure the code dependencies work.

- [x] Ensure the etcd client works with a locally spun up etcd instance (docker)
- [x] kubeadm tests This will become a whole lot simpler after #31. For now:
    - [x] CreatePKI() - Internally `kubeadm phase cert...`
    - [x] SaveAssets() - as if primary master
    - [x] LoadAndSerializeAssets() - as if secondary master
    - [x] WriteManifests()
    - [x] CreateKubeConfig() - Internally `kubeadm phase kubecfg...`
- ~~kmm/kubelet - requires systemd service (may use for e2e test also)~~ - will revisit after #35 (will probably build in another container.

Out of scope (for this ticket):
- kube api dependant tests - see e2e https://github.com/UKHomeOffice/keto-k8/issues/35
- etcd certs generate - this will probably be broken out into separate tool as this feature has a different release cadence and dependency list to the rest of the tool.

### PR DESCRIPTION
**Pull request by lewismarshall**
So far added etcd.

Will Fix #40 
Also Fixes #9

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -16,6 +18,12 @@ pipeline:
       - golint $(go list ./... | grep -v /vendor | grep -v /mocks)
       - go vet $(go list ./... | grep -v /vendor)
       - go get github.com/vektra/mockery/.../
+      # We need kubeadm and docker for integration tests
```
Comment (vaijab) [inline .drone.yml]: Have below as a script instead maybe?

File: pkg/etcd/etcd_test.go [hunk 2]
```diff
@@ -1,3 +1,219 @@
...
+	"os/exec"
+	"testing"
+
+	"github.com/UKHomeOffice/keto-k8/pkg/fileutil"
+	"time"
+)
+
+const containerName string = "ectd_int_test"
+
+var etcdStarted = false
```
Comment (vaijab) [inline pkg/etcd/etcd_test.go]: I would not use global variables in testing. Test cases should be self contained without state leakage.

Comment (lewismarshall) [inline pkg/etcd/etcd_test.go]: They are, the etcd back end is only loaded once for speed and no tests use overlapping data.

Comment (lewismarshall) [inline pkg/etcd/etcd_test.go]: I do agree in principle but it's a trade-off I guess but for now speed over safeguarding future poor tests.

Comment (lewismarshall) [inline pkg/etcd/etcd_test.go]: I do agree in principle but it's a trade-off I guess but for now speed over safeguarding future poor tests.

Comment (lewismarshall) [inline pkg/etcd/etcd_test.go]: I do agree in principle but it's a trade-off I guess but for now speed over safeguarding future poor tests.

Comment (lewismarshall) [inline pkg/etcd/etcd_test.go]: This is tricky on a phone!

File: .drone.yml (hunk 1 above)
Comment (lewismarshall) [inline .drone.yml]: By your command :)

**Pull request merged**